### PR TITLE
Implemented test to create a discovered_host via uploading facts

### PR DESCRIPTION
**Pull request by sghai**
Added a test to create discovered host. For now, I defined a fn `_create_discovered_host` to create discovered host with required facts. We can add more facts later in same function. 

The idea behind defining the `_create_discovered_host` to use it for further tests like `test_delete_discovered_host`.

### CODE REVIEW

File: tests/foreman/api/test_discovery.py [hunk 1]
```diff
@@ -1,9 +1,36 @@
...
+    """Creates discovered host by uploading few fake facts.
+
+    :param str name: Name of discovered host. If ``None`` then a random
+        value will be generated.
+    :param str ipaddress: A valid ip address.
+        If ``None`` then then a random value will be generated.
+    :return: A ``DiscoveredHost`` object.
+
+    """
+    if name is None:
```
Comment (elyezer) [inline tests/foreman/api/test_discovery.py]: What about defaulting name to None? This way we can call the function without any argument, if we are forcing the user to provide the value, then we should not generate a random value if decided to pass None.

**Comment by sghai** [comment]
Test result:

```
@Test: Upload fake facts to create a discovered hosts ... ok

----------------------------------------------------------------------
Ran 1 test in 7.827s

OK
```

File: tests/foreman/api/test_discovery.py (hunk 1 above)
Comment (sghai) [inline tests/foreman/api/test_discovery.py]: Yeah.. agreed. We don't want force user to provide name. I think `defaulting name to None` would be a good idea.

File: tests/foreman/api/test_discovery.py [hunk 2]
```diff
@@ -1,9 +1,36 @@
...
+        name = gen_string('alpha')
+    if ipaddress is None:
+        ipaddress = gen_ipaddr()
+    if discovery_bootif is None:
+        discovery_bootif = gen_mac()
+    return entities.DiscoveredHost().facts(
```
Comment (elyezer) [inline tests/foreman/api/test_discovery.py]: The below looks more readable:

```python
return entities.DiscoveredHost().facts({
    u'facts': {
        u'name': name,
        u'ipaddress': ipaddress,
        u'discovery_bootif': discovery_bootif,
    }
})
```

**Comment by elyezer** [comment]
ACK pending comment

File: tests/foreman/api/test_discovery.py (hunk 2 above)
Comment (sghai) [inline tests/foreman/api/test_discovery.py]: :+1: 

I always confused with formatting. Thank for correcting it.

File: tests/foreman/api/test_discovery.py [hunk 3]
```diff
@@ -1,9 +1,38 @@
...
+    :return: A ``DiscoveredHost`` object.
+
+    """
+    if name is None:
+        name = gen_string('alpha')
+    if ipaddress is None:
+        ipaddress = gen_ipaddr()
+    if discovery_bootif is None:
+        discovery_bootif = gen_mac()
+    return entities.DiscoveredHost().facts({
```
Comment (Ichimonji10) [inline tests/foreman/api/test_discovery.py]: Add `data={`:

```python
return entities.DiscoveredHost().facts(data={
```

The current statement is equivalent to the following:

```python
return entities.DiscoveredHost().facts(synchronous={
```

File: tests/foreman/api/test_discovery.py [hunk 4]
```diff
@@ -1,9 +1,38 @@
...
 
 
+def _create_discovered_host(name=None, ipaddress=None, discovery_bootif=None):
+    """Creates discovered host by uploading few fake facts.
+
+    :param str name: Name of discovered host. If ``None`` then a random
+        value will be generated.
+    :param str ipaddress: A valid ip address.
+        If ``None`` then then a random value will be generated.
+    :return: A ``DiscoveredHost`` object.
```
Comment (Ichimonji10) [inline tests/foreman/api/test_discovery.py]: The server's response (likely a dict) is returned. A `DiscoveredHost` object is not returned.

**Comment by Ichimonji10** [comment]
I ran tests with this change applied:

```diff
diff --git a/tests/foreman/api/test_discovery.py b/tests/foreman/api/test_discovery.py
index 6c222c0..0154cd3 100644
--- a/tests/foreman/api/test_discovery.py
+++ b/tests/foreman/api/test_discovery.py
@@ -23,7 +23,7 @@ def _create_discovered_host(name=None, ipaddress=None, discovery_bootif=None):
         ipaddress = gen_ipaddr()
     if discovery_bootif is None:
         discovery_bootif = gen_mac()
-    return entities.DiscoveredHost().facts({
+    return entities.DiscoveredHost().facts(data={
         u'facts': {
             u'name': name,
             u'ipaddress': ipaddress,
```

Results:

```
$ nosetests tests/foreman/api/test_discovery.py -m test_upload_facts_for_discovered_host
......
----------------------------------------------------------------------
Ran 6 tests in 2.480s

OK
```

ACK pending comments.

**Comment by sghai** [comment]
Thanks @Ichimonji10. I've updated the PR. thanks

**Comment by Ichimonji10** [comment]
ACK

File: tests/foreman/api/test_discovery.py [hunk 5]
```diff
@@ -1,9 +1,38 @@
...
 
 
+def _create_discovered_host(name=None, ipaddress=None, discovery_bootif=None):
+    """Creates discovered host by uploading few fake facts.
+
+    :param str name: Name of discovered host. If ``None`` then a random
+        value will be generated.
+    :param str ipaddress: A valid ip address.
+        If ``None`` then then a random value will be generated.
+    :return: A ``dict`` of ``DiscoveredHost`` facts.
```
Comment (Ichimonji10) [inline tests/foreman/api/test_discovery.py]: :+1:

**Pull request merged**